src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -35,13 +37,33 @@ public override bool ShouldOverrideUrlLoading(AWebView? view, IWebResourceReques
 		{
 			// Handle redirects to the app custom scheme by reloading the URL in the view.
 			// Handle navigation to external URLs using the system browser, unless overriden.
-			var requestUri = request?.Url?.ToString();
+			if (request?.Url is { } url)
+			{
+				if (url.Scheme is "http" or "https")
+				{
+					if (TryOverrideHttpUrlLoading(view, request, url.ToString()))
+					{
+						return true;
+					}
+				}
+				else
+				{
+					TryStartActivityFromUri(url.ToString(), uriHasIntentScheme: url.Scheme == "intent");
```

Comment:
Likewise here. If we passed it as an `Android.Net.Uri`, we wouldn't need the `uriHasIntentScheme` param as the receiving code could encapsulate this check.